collector/rapl_linux.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !norapl
+
+package collector
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+)
+
+type raplCollector struct {
+	fs sysfs.FS
+}
+
+func init() {
+	registerCollector("rapl", defaultEnabled, NewRaplCollector)
+}
+
+// NewRaplCollector returns a new Collector exposing RAPL metrics.
+func NewRaplCollector() (Collector, error) {
+	fs, err := sysfs.NewFS(*sysPath)
+	if err == nil {
```

Comment:
Please invert the logic here to make it more idiomatic go, e.g:
```
if err != nil {
  return nil, err
}
```

Reply:
Done, and thanks for reviewing!